### PR TITLE
Add student hub type and doc the feature

### LINKED ISSUE #164: Student Hubs
### Summary

Adding Student Hubs Channel

### What is the feature request for?

The core library

### The Problem

Currently student hubs are not included in the libary, but are already available. this might result in errors regarding invites trying to be resolved (bots can not join so far but invites might throw errors)

### The Ideal Solution

Implement a basic placeholder for them without any major functions

### The Current Solution

none

### Additional Context

https://github.com/discord/discord-api-docs/pull/3757

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

closes #164 

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by veni-vidi-code** [comment]
ChannelType will also need the addition of 14 https://github.com/nextcord/nextcord/blob/b644c29841ec878b77b6a38bdb6e754bf7ec90e2/nextcord/types/channel.py#L41

**Review by veni-vidi-code — CHANGES_REQUESTED** [review]
ChannelType will also need the addition of 14 https://github.com/nextcord/nextcord/blob/b644c29841ec878b77b6a38bdb6e754bf7ec90e2/nextcord/types/channel.py#L41

File: nextcord/types/channel.py [hunk 1]
```diff
@@ -155,3 +155,6 @@ class StageInstance(TypedDict):
     topic: str
     privacy_level: PrivacyLevel
     discoverable_disabled: bool
+
+class GuildDirectory:
+    type: Literal[14]
```
Comment (veni-vidi-code) [inline nextcord/types/channel.py:160]:
```suggestion
    type: Literal[14]

```

Comment (veni-vidi-code) [reply nextcord/types/channel.py:160]: would be nice to have a newline at the end, pep8

Comment (ooliver1) [reply nextcord/types/channel.py]: Done

**Comment by ooliver1** [comment]
Hang on GitHub doesn't seem to understand what a newline is when I select commit to branch

**Comment by ooliver1** [comment]
stalled as wontbe merged until in ddocs

**Comment by ooliver1** [comment]
it doesnt sound like itll be merged for ddocs any time soon

**Comment by vcokltfre** [comment]
I've converted this to a draft as it is awaiting ddocs update

**Comment by ooliver1** [comment]
seems like they may never merge anything in ddocs

**Pull request closed without merging**